napari/utils/colormaps/colormap_utils.py

```diff
@@ -427,25 +485,30 @@ def label_colormap(num_colors=256, seed=0.5):
     """
     # Starting the control points slightly above 0 and below 1 is necessary
     # to ensure that the background pixel 0 is transparent
-    midpoints = np.linspace(0.00001, 1 - 0.00001, num_colors + 1)
-    control_points = np.concatenate(([0], midpoints, [1.0]))
+    midpoints = np.linspace(0.00001, 1 - 0.00001, num_colors - 1)
+    control_points = np.concatenate([[0], midpoints, [1.0]])
     # make sure to add an alpha channel to the colors
-    colors = np.concatenate(
-        (
-            _color_random(num_colors + 2, seed=seed),
-            np.full((num_colors + 2, 1), 1),
-        ),
-        axis=1,
-    )
-    # Insert alpha at layer 0
-    colors[0, :] = 0  # ensure alpha is 0 for label 0
+    if num_colors == 49:
+        colors = np.array(CLASSICAL_LABEL_COLORMAP)
+    else:
+        colors = np.concatenate(
+            (
+                _color_random(num_colors),
+                np.full((num_colors, 1), 1),
+            ),
+            axis=1,
+        )
+
+    colors = colors[
+        np.random.default_rng(int(seed * 1000)).permutation(num_colors)
+    ]
```

Comment:
Ok so this is totally changing the source of randomness from the `_color_random` function to the permutation — with `_color_random` now fixed for a particular seed (what seed?). And this has to be this way because we no longer regenerate the colours, we just shuffle them, so the first shuffling needs to be consistent?

In that case I might suggest changing the API and postponing this improvement to 0.5.0... What do you think? This `* 1000` is just really weird for me. The API could just be the new `rng` keyword that [SciPy is adopting](https://github.com/scientific-python/specs/pull/180) (to my dismay, but we should follow the community).

Alternatively, we could *add* the `rng=` keyword and deprecate `seed`. In the meantime, keep using seed for `_color_random` and use `rng=` for the shuffling...? (Maybe even with `rng=None` signifying no shuffling — and thus backwards compatibility.)